src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,130 @@
+     * It will listen to the incoming users's command and process the comment accordingly.
+     */
+
+    public void run() {
+        this.ui.greetLogo();
+        this.ui.greet();
+        ui.printInputRequest();
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            try {
+                String input = sc.nextLine();
+                input = input.toLowerCase();
+                if (input.equalsIgnoreCase("bye")) {
+                    this.ui.bye();
+                    break;
+                } else if (input.equalsIgnoreCase("list")) {
+                    this.taskList.list();
+                } else if (input.startsWith("done")) {
+                    try {
+                        int taskNumber = Integer.parseInt(input.substring(5));
+                        this.taskList.done(taskNumber, this.storage);
+                    } catch (StringIndexOutOfBoundsException e) {
+                        System.out.println(new DukeException("OOPS!!! Done format is wrong"));
+                    } catch (NumberFormatException e) {
+                        System.out.println(new DukeException("OOPS!!! Done format is wrong"));
+                    }
+                } else if (input.startsWith("delete")) {
+                    try {
+                        int taskNumber = Integer.parseInt(input.substring(7));
+                        this.taskList.delete(taskNumber, this.storage);
+                    } catch (StringIndexOutOfBoundsException e) {
+                        System.out.println(new DukeException("OOPS!!! Delete format is wrong"));
+                    } catch (NumberFormatException e) {
+                        System.out.println(new DukeException("OOPS!!! Delete format is wrong"));
+                    }
+                } else if (input.startsWith("find")) {
+                    String[] keyWords = input.substring(5).trim().split(" ");
+
+                    FindTask findTask = new FindTask(this.taskList.getTaskList());
+                    findTask.search(keyWords);
+                    findTask.list();
+
+                } else {
+                    if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+
+                        if (input.startsWith("todo")) {
+                            this.taskList.addTodo(input, this.storage);
+                        } else if (input.startsWith("deadline")) {
+                            this.taskList.addDeadline(input, this.storage);
+                        } else if (input.startsWith("event")) {
+                            this.taskList.addEvent(input, this.storage);
+                        }
+                    } else {
+                        throw new DukeException("      ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                    }
+                }
+            } catch (DukeException e) {
+                System.out.println(e);
+            }
+            ui.printInputRequest();
```

Comment:
Same here! I think you're missing a "this." here. Not that it'll affect the code but just for the sake of uniformity and consistency!